h/static/scripts/controllers/confirm-submit-controller.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+var Controller = require('../base/controller');
+
+/* Turn a normal submit element into one that shows a confirm dialog.
+ *
+ * The element's form will only be submitted if the user answers the
+ * confirmation dialog positively.
+ *
+ */
+class ConfirmSubmitController extends Controller {
+  constructor(element, options) {
+    super(element);
+
+    var window_ = options.window || window;
```

Comment:
This is a "test seam" (I saw this done with `window` elsewhere in the code as well, guessing by Rob)